src/btree/bt_read.c

```diff
@@ -700,6 +700,9 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
             }
             goto read;
         case WT_REF_DISK:
+            /* Read internal pages if requested */
+            if (LF_ISSET(WT_READ_INT_PAGE) && ((WT_ADDR *)ref->addr)->type == WT_ADDR_INT)
+                goto read;
```

Comment:
Why can't we be here with `WT_REF.addr` pointing to an on-page cell?

Reply:
I didn't clearly understand the comment. Maybe I can explain what I am thinking, let me know if any problem.

The above code is to ensure to read the internal pages that are not in memory for the history store.
This is done when the parent page is getting processed, all the `pindex` entries are processed one by one. Only the pages that are not in memory will come to this case to let them load only when they are internal pages.

Reply:
Got it, this makes sense now. Again, I'd suggest creating a function like `__ref_is_leaf()` that returns if the page is an internal page.

Reply:
Thanks, used `__ref_is_leaf()` function to find out the page is leaf or internal.